src/sentry/tasks/deletion.py

```diff
@@ -16,6 +17,22 @@
 MAX_RETRIES = 5
 
 
+@instrumented_task(
+    name="sentry.tasks.deletion.reattempt_deletions", queue="cleanup", acks_late=True
+)
+def reattempt_deletions():
+    from sentry.models import ScheduledDeletion
+
+    # If a deletion is in progress and was scheduled to run more than
+    # a day ago we can assume the previous job died/failed.
+    # Turning off the in_progress flag will result in the job being picked
+    # up in the next deletion run allowing us to start over.
+    queryset = ScheduledDeletion.objects.filter(
+        in_progress=True, aborted=False, date_scheduled__lte=timezone.now() - timedelta(days=1)
```

Comment:
When would the `aborted` flag be true?

Reply:
Right now it can't. ModelDeletion implementations can raise `DeleteAborted` when the record they are supposed to delete has been undeleted, but nothing handles that error and converts it into an update to the `ScheduledDeletion`. I'm planning on fixing that up today.